### DISCUSSION TITLE
StoneDB Bug Bounty Program Season 1

### DISCUSSION THREAD

**Discussion by Yeekin-GYJ** [stoneatom/stonedb#385] [Events]
Our dearest users and community members,
Thank you for staying with us since our official release and open source in June 20th. Now, our first community activity — Bug Bounty Program — is coming.
Though StoneDB is well prepared, bugs are inevitable, and we need you to help catch them.
The first season of the Bug Bounty Program focuses on our docs. The only thing you need to do is to catch the bugs hiding in our docs and report them by issue or pull request (PR), and then you will score the corresponding points. You can use the points to exchange delicate peripheral prizes. And of course, you will be in our contributor list if you're currently not.
To participate in the activity, you only need to perform the following two steps:

### Step 1. Sign up
Join in our [Slack channel](https://stonedb.slack.com/ssb/redirect#/shared-invite/email) and send a message in the format of "[Your GitHub username]: sign up for the Bug Bounty Program". Then, you're in.

### Step 2. Report bugs

**Submit issues**
Read our [docs](https://github.com/stoneatom/stonedb/tree/stonedb-5.7/Docs) and submit an issue whenever you catch a bug. Before creating an issue, please check whether the issue already exists in the[ issue](https://github.com/stoneatom/stonedb/issues) list. If no, please create an issue in the [stonedb](https://github.com/stoneatom/stonedb) library.

**Commit PRs**
After you catch a bug and create an issue, fork the [stonedb](https://github.com/stoneatom/stonedb) library, fix the issue, and then commit the PR. Besides the issues that you create, you can also fix issues that are pending for solution.
Addresses of doc libraries
**Link to English docs:** https://github.com/stoneatom/stonedb/tree/stonedb-5.7/Docs
**Link to Chinese docs:** https://github.com/stoneatom/stonedb/tree/stonedb-5.7/website/i18n/zh/docusaurus-plugin-content-docs/current
This season of our Bug Bounty Program is specific for docs of the StoneDB-5.7 branch.

### Scoring rules
We classify bugs based on severity levels: P01 indicates low-severity, P02 medium-severity, P03 high-severity, and P04 critical.

**P01: 1 point/issue, 2 points/PR**
Low-severity bugs refer to minor errors such as:
● Typos, spelling errors, invalid links, and formatting errors
● Readability issues that do not affect understandings

**P02: 3 points/issue, 5 points/PR**
Medium-severity bugs are usability issues, including:
● Lack of or misleading key steps which cannot guide users to complete specific tasks
● Lack of necessary prerequisites or precautions
● Code errors, such as missing spaces or quotation marks

**P03: 5 points/issue, 8 points/PR**
High-severity bugs are accuracy issues, including:
● Ambiguous descriptions
● Errors in architecture figures, diagrams that illustrate working principles, and feature descriptions
● Incorrect descriptions of key parameters, which will result in operation failures

**P04: 8 points/issue, 12 points/PR**
Critical bugs are defects that may block functionality of an entire system or module, for example,
● Lack of notifications for risky operations that can damage key data or interrupts the system running

### Exchange points
1. Available prizes include: badge = 5 points, polo shirt = 10 points, umbrella = 15 points, backpack = 20 points (If you get enough points, you can exchange more than one prize.)
If you live outside of China, we may fail to send you physical prizes due to logistics restrictions. Please don’t feel upset. We also prepare certificates of honor and thank-you letters and we’ll send them to you by email.

2. Once your issue or PR passes the review, the corresponding number of points are accumulated in your account. When your points are sufficient to exchange your desired prizes, contact our assistant and then we'll send you the corresponding prizes. If you have any problems about this activity, you can contact us in our [Slack channel](https://stonedb.slack.com/ssb/redirect#/shared-invite/email) or participate in the [discussion](https://github.com/stoneatom/stonedb/discussions).

There are still some empty topics waiting for your contribution. A complete topic can be used to exchange a full set of peripheral prizes (including a badge, polo shirt, umbrella, and backpack).
![image](https://user-images.githubusercontent.com/101704468/183631355-696b95e8-73d3-49d5-91b6-5ef9628456e7.png)

### A few more rules
1. Use one issue or PR to report or fix only one bug.
2. Duplicate bugs are regarded as one bug even though they appear in different parts of the docs.
3. If more than one contributor reports the same bug, only the first contributor gets the points.
4. Label your issue or PR based on the severity level (P01, P02, P03, or P04) to help improve our efficiency in reviewing your issue or PR and counting your points.
![image](https://user-images.githubusercontent.com/101704468/183631421-9ddaae5b-f620-46b7-bbdd-c740ca58ae35.png)

5. Use English for all kinds of communications on GitHub. Issues and PRs in other languages will be regarded as invalid and our assistant will remind you to rewrite.

6. The following types of issues will be considered as invalid:
  ○ Issues that cannot be located
  ○ Issues that cannot be reproduced
  ○ Issues that already exists
  ○ Other issues that fail to pass the review of PMC members

### Recommendations
Community website: https://stonedb.io/
Code library: https://github.com/stoneatom/stonedb
![image](https://user-images.githubusercontent.com/101704468/183631764-c9dd5913-f8d3-4d80-9966-d13a36d3af48.png)

**Comment by RingsC** [comment]
Anything improvements are welcome. Come on everybody!!!

**Comment by Nliver** [comment]
Welcome to find bugs together~

https://github.com/stoneatom/stonedb/issues/new?assignees=&labels=A-documentation&template=documentation.yml&title=docs%3A+

**Comment by davidshiz** [comment]
welcome
